app/controllers/v1/sessions_controller.rb

```diff
@@ -138,15 +121,47 @@ def user_login(saml_response)
       @current_user, @session_object = user_session_form.persist
       set_cookies
       after_login_actions
-      redirect_to url_service(user_session_form.saml_uuid).login_redirect_url
-      if location.start_with?(url_service.base_redirect_url)
-        # only record login stats if the user is being redirect to the site
-        # some users will need to be up-leveled and this will be redirected
-        # back to the identity provider
-        login_stats(:success, saml_response, user_session_form)
+      helper = url_service(user_session_form.saml_uuid)
+      if helper.should_uplevel?
+        render_login('verify')
+      else
+        redirect_to helper.login_redirect_url
+        if location.start_with?(helper.base_redirect_url)
```

Comment:
I think this if block and the one on line 125 are testing the same thing.  Now that there is an else block I think this can be removed

Reply:
actually I just realized there is a bug with this if statement that makes the if on line 125 preferred.  When we redirect users to cerner, this block won't get executed.

Reply:
Ooh, good catch.